resources/knative-eventing/charts/knative-eventing/templates/pre-upgrade-job.yaml

```diff
@@ -48,50 +51,44 @@ data:
 
-    if kubectl get crd subscriptions.eventing.knative.dev 2>/dev/null; then
-      echo "deleting subscriptions"
-      kubectl delete subscriptions.eventing.knative.dev --all -n kyma-system --ignore-not-found
-    fi
-    if kubectl get crd channels.eventing.knative.dev 2>/dev/null; then
-      echo "deleting channels"
-      kubectl delete channels.eventing.knative.dev --all -n kyma-system --ignore-not-found
-    fi
-    if kubectl get crd clusterchannelprovisioners.eventing.knative.dev 2>/dev/null; then
-      echo "deleting clusterchannelprovisioners"
-      kubectl delete clusterchannelprovisioners.eventing.knative.dev --all --ignore-not-found
-    fi
-
-    #### Remove CRDs
-    # pre 0.10 CRDs, deprecated
-    kubectl delete --ignore-not-found crd \
-      subscriptions.eventing.knative.dev \
-      clusterchannelprovisioners.eventing.knative.dev \
-      channels.eventing.knative.dev
-    # CRDs previously created by the pre-upgrade hook, now part of the eventing deployment manifests
-    kubectl delete --ignore-not-found crd \
-      parallels.messaging.knative.dev \
+    echo "+ Deleting outdated CRDs"
+    crds=(
+      # CRDs previously created by the pre-upgrade hook, now part of the eventing deployment manifests.
+      # *DELETE THE LINES BELOW BEFORE THE NEXT KYMA RELEASE** (https://github.com/kyma-project/kyma/issues/7387)
+      parallels.messaging.knative.dev
       subscriptions.messaging.knative.dev
+    )
+    for crd in "${crds[@]}"; do
+      kubectl delete --ignore-not-found "crd/${crd}"
+      while kubectl get "crd/${crd}" &>/dev/null; do
+        echo "waiting for deletion of crd/${crd}"
+        sleep 1
+      done
+    done
+
+    echo "+ Stopping event-bus Subscription controller"
+    # adds/removes a finalizer to Knative Subscriptions, hence needs to be stopped *after* Knative Subscriptions are deleted.
+    if kubectl -n kyma-system get deploy/event-bus-subscription-controller &>/dev/null; then
+      kubectl -n kyma-system scale deploy/event-bus-subscription-controller --replicas=0
+    fi
 
-    # MutatingWebhookConfiguration previously created by the pre-upgrade hook, now part of the eventing deployment manifests
+    # MutatingWebhookConfiguration previously created by the pre-upgrade hook, now part of the eventing deployment manifests.
+    echo "+ Deleting outdated MutatingWebhookConfigurations"
     kubectl delete --ignore-not-found mutatingwebhookconfiguration \
       webhook.eventing.knative.dev
 
-    # Remove auto-generated Secret.
+    echo "+ Deleting auto-generated Secrets"
     # Avoids update conflict: "Secret with the name "eventing-webhook-certs" already exists in the cluster and wasn't defined in the previous release."
     kubectl -n knative-eventing delete --ignore-not-found \
       secret/eventing-webhook-certs
 
-    #### deploy/event-bus-subscription-controller adds/removes a finalizer to knative subscriptions hence it needs to be deleted after knative subscriptions are deleted
```

Comment:
What was the problem with deletion of the deployment? I remember, we decided not to scale down earlier as it was somewhat hacky.

Reply:
Whatever we do is gonna be hacky since we operate on a controller managed by a different chart. If I have to chose, I find scaling more graceful than wiping objects.

Reply:
Not sure why you said something different here, https://github.com/kyma-project/kyma/pull/6072#discussion_r337361756

Reply:
This comment was made with the assumption our `kubectl` version waits until objects are deleted. As you can see in the PR, it doesn't. `kubectl` doesn't wait for scaling either, so the only real solution here is to upgrade our `kubectl` image.

Reply:
`delete` restored. Used the `--wait` kubectl flag.